miniapps/gslib/findpts.cpp

```diff
@@ -46,23 +47,63 @@
 //    findpts -m ../../data/tinyzoo-3d.mesh -o 1 -mo 1
 
 #include "mfem.hpp"
+#include "../common/mfem-common.hpp"
 
 using namespace mfem;
 using namespace std;
 
+// Experimental - class used to update GridFunction post p-refinement.
+// Can change as support for p-refinement evolves.
+class PRefinementGFUpdate
+{
+private:
+   FiniteElementSpace *src;
+
+public:
+   /// @brief Used to Update GridFunction post p-refinement.
+   /// Initialize with FESpace prior to p-refinement.
+   PRefinementGFUpdate(const FiniteElementSpace& src_);
+
+   /// Destructor
+   ~PRefinementGFUpdate();
+
+   /// @brief Update GridFunction using PRefinementTransferOperator.
+   /// Do not use GridFunction->Update() prior to this method as it is
+   /// handled internally.
+   void GridFunctionUpdate(GridFunction &targf);
+};
+
+PRefinementGFUpdate::PRefinementGFUpdate(const FiniteElementSpace &src_)
+{
+   src = new FiniteElementSpace(src_);
+}
+
+PRefinementGFUpdate::~PRefinementGFUpdate()
+{
+   delete src;
+}
+
+void PRefinementGFUpdate::GridFunctionUpdate(GridFunction &targf)
+{
+   MFEM_VERIFY(targf.GetSequence() != targf.FESpace()->GetSequence(),
+               "GridFunction should not be updated prior to UpdateGF.");
+   Vector srcgf = targf;
+   targf.Update();
+   PRefinementTransferOperator preft =
+      PRefinementTransferOperator(*src, *(targf.FESpace()));
```

Comment:
Should `src` be replaced with the new space, in case this function is called again?

Reply:
I added the function to explicitly set the source space.. we don't want to automatically replace it because the user might want to update multiple grid functions.